### PR TITLE
Change tests to use `Vec<u8>` instead of `String`

### LINKED ISSUE #1916: Tests: Replace `String` with `Vec<u8>` for stdout (stderr?)
Currently, the `CmdResult` struct has stdout as:
```rust
    /// captured utf-8 standard output after running the Command
    pub stdout: String,
 ```
This means that we can't completely test any utility that should be able to handle arbitrary bytes, for no added benefit.
The same holds true for stderr, which can hold binary data, but I don't know whether there are any legitimate uses for this.

As far as I can tell the implementation should be easy. This is the code of the `run` function:
```rust
pub fn run(&mut self) -> CmdResult {
        let prog = self.run_no_wait().wait_with_output().unwrap();

        CmdResult {
            tmpd: self.tmpd.clone(),
            code: prog.status.code(),
            success: prog.status.success(),
            stdout: from_utf8(&prog.stdout).unwrap().to_string(),
            stderr: from_utf8(&prog.stderr).unwrap().to_string(),
        }
    }
```
where stdout and stderr are needlessly converted. (ofc changing this will affect some other parts of the code)

I'd implement this once I'm done with #1911, if no one else shows interest

### PR DESCRIPTION
**Pull request by Mikadore**
This PR changes the internals of `CmdResult` to work with a vector of bytes instead of a string.
It adds methods to the public API to support this where applicable. In addition to this, it makes these fields private to properly encapsulate the `CmdResult` struct. All tests were adjusted accordingly when needed.
Closes #1916

### CODE REVIEW

**Comment by sylvestre** [comment]
I am sorry but this PR cannot be merged now.
You are mixing too many different things into the same PR and there is too many different patches in this PR to merge as it.

Could you please it into different PR? thanks

**Comment by sylvestre** [comment]
git format-patch -68
and then apply patches into different branches and then push :)

**Comment by Mikadore** [comment]
> I am sorry but this PR cannot be merged now.
> You are mixing too many different things into the same PR and there is too many different patches in this PR to merge as it.
> 
> Could you please it into different PR? thanks

Most of these changes are required in order to even build the utils.
The problem is that many tests directly access `stdout`/`stderr`, even if `stdout_is` would suffice etc.

The only way forward without changing all at once is to abstract away `stdout`/`stderr` (a .stdout_contains` would be nice for example). So I'll do that, and convert tests one - by - one.

File: tests/by-util/test_date.rs [hunk 1]
```diff
@@ -8,15 +8,13 @@ use rust_users::*;
 #[test]
 fn test_date_email() {
     let (_, mut ucmd) = at_and_ucmd!();
-    let result = ucmd.arg("--rfc-email").run();
-    assert!(result.success);
+    ucmd.arg("--rfc-email").succeeds();
```
Comment (sylvestre) [inline tests/by-util/test_date.rs:11]: this change seem unrelated, no?

Comment (Mikadore) [reply tests/by-util/test_date.rs:11]: Well kinda, since we're making parts of the struct private, and the code would've needed to be refactored, either way, I thought to do this anyway. It also makes lifetimes less of a pain on that branch.
